db/migrate/20210119122653_add_flat_view.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require './lib/views_schema'
+
+# Flattened view of the events database with one row per subject.
+# Metadata is excluded
+class AddFlatView < ActiveRecord::Migration[5.2]
+  def up
+    ViewsSchema.create_view(
+      'flat_events_view',
+      <<~SQL
+        SELECT
+          events.id AS wh_event_id,
+          events.uuid AS event_uuid_bin,
+          INSERT(INSERT(INSERT(INSERT(LOWER(HEX(events.uuid)),9,0,'-'),14,0,'-'),19,0,'-'),24,0,'-') AS event_uuid,
```

Comment:
😆

Reply:
I'll check it on UAT to see it's doing the right thing.

Reply:
Annoyingly MySQL 8 has BIN_TO_UUID which would be *much* nicer.

Reply:
Interesting, I wonder when we should consider upgrading?

Reply:
I've asked the DBAs. We're on the list of people who they'll look at rolling out the upgrade first. Although given the number of users of the warehouses, we'll probably want to do those a little later in the upgrade process.

Last time I spoke to them they didn't have a road-map set out for the update.

Lots of nice stuff in MySQL 8 though.